### DISCUSSION TITLE
Self-Host: Replication - Force a new replication

### DISCUSSION THREAD

**Discussion by jpg918** [ente-io/ente#4428] [Q&A] [answered]
Hello,

First, thank you for this wonderful app!  I have been searching for this for quite some time to help de-couple from prying, corporate eyes.  I greatly appreciate the ability to self-host.

My question is related to replication.  I initially configured Museum and MinIO (minio-1) on same host via docker compose and everything is working great in a sandbox.  I am now experimenting with replication.  I have another host running MinIO (minio-2) and have mapped data in these buckets successfully to a NAS.  

I enabled replication, pointed buckets 2 and 3 to the minio-2 instance, and then started museum.  I see replication is kicked off in the logs and everything from the minio-1 bucket 1 instance has replicated successfully to minio-2 to buckets 2 and 3. 

I am attempting to understand and try a MinIO (minio-3) instance for the 3rd bucket. But once this 3rd instance is configured and set up and museum pointed at this minio-3 for 3rd bucket (future cloud S3), I don't see data being replicated to the newly configured bucket.

My assumption is that museum sees that it has already replicated the data when I first enabled replication. 

My question, is there a process to force replication again?  I've disabled replication and reenabled but no luck.  Looking for some help!

Thank you in advance!

**Comment by mnvr** [comment, marked as answer]
> museum sees that it has already replicated the data when I first enabled replication.

Right. This state is kept in the postgres DB, so if you want to trigger replication again you will need to tinker with the DB. There isn't an API for doing it (we'll try to add one), meanwhile you might be able to get museum to replicate again by resetting the `scw`/`want_scw` (or `wasabi`/`want_wasabi`, depending on the bucket under consideration) columns in the `object_copies` table. I'm mentioning all this just from the top of my head in case it help, this might not work or break other invariants, so be careful, keep a backup, and test it out on a smaller sample first.

(the `scw` column tracks when the file was replicated to that bucket, and the `want_scw` column tells museum that it should replicate it there. the state we want to reset things to is - set `want_scw` to true, and nil out `scw`. ditto for the wasabi pair)

**Reply by jpg918** [reply]
Thanks @mnvr for the quick response, it put me on the right path!

One update I had to make was to set the '_last_attempt_' column to '0' as well as the column pair updates you mentioned.  Once I made this change, I started Museum and the rows/objects I expected were re-replicated to the buckets where I cleared the values as I hoped.

I was able to re-replicate 1 bucket, or both, depending on the columns updated within the _object_copies_ table.

Samples I tested:

```
UPDATE public.object_copies
SET  scw=NULL, want_scw=true, last_attempt=0
Where object_key like '1580559962386438/%';
```
```
UPDATE public.object_copies
SET  wasabi=NULL, want_wasabi=true, last_attempt=0
Where object_key like '1580559962386438/%';
```
```
UPDATE public.object_copies
SET  scw=NULL, want_scw=true, wasabi=NULL, want_wasabi=true, last_attempt=0
Where object_key like '1580559962386438/%';
```
I see some rows in the _object_copies_ table where the values are always 'false', and I noted in the logs that these are ignored by replication because they are marked for delete or deleted? 

Will these rows ever clean up? Or are they required to be kept?

**Reply by mnvr** [reply]
Great that you got it working!

> Will these rows ever clean up? Or are they required to be kept?

They'll clean up when the object gets deleted, I'd let them be.
